File: pkg/codegen/testing/test/testdata/simple-data-types-pp/go/simple-data-types.go
package main

import (
	"fmt"
	"io/ioutil"

	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func readFileOrPanic(path string) pulumi.StringPtrInput {
	data, err := ioutil.ReadFile(path)
	if err != nil {
		panic(err.Error())
	}
	return pulumi.String(string(data))
}

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		basicStrVar := "foo"
		ctx.Export("strVar", pulumi.String(basicStrVar))
		ctx.Export("computedStrVar", pulumi.String(fmt.Sprintf("%v/computed", basicStrVar)))
		ctx.Export("strArrVar", pulumi.ToStringArray([]string{
			"fiz",
			"buss",
		}))
		ctx.Export("intVar", pulumi.Float64(42))
		ctx.Export("intArr", pulumi.ToFloat64Array([]float64{
			1,
			2,
			3,
			4,
			5,
		}))
		ctx.Export("readme", readFileOrPanic("./Pulumi.README.md"))
		return nil
	})
}

File: pkg/codegen/testing/test/testdata/simple-data-types-pp/python/simple-data-types.py
import pulumi

basic_str_var = "foo"
pulumi.export("strVar", basic_str_var)
pulumi.export("computedStrVar", f"{basic_str_var}/computed")
pulumi.export("strArrVar", [
    "fiz",
    "buss",
])
pulumi.export("intVar", 42)
pulumi.export("intArr", [
    1,
    2,
    3,
    4,
    5,
])
pulumi.export("readme", (lambda path: open(path).read())("./Pulumi.README.md"))
